### PR TITLE
ImageMatrix Initialization Tests

### PR DESCRIPTION
**Pull request by JesseMckinzie**
Added tests to check if the ImageMatrix has been initialized for all features calculations.

### CODE REVIEW

File: tests/test_main_nyxus.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            roidata.raw_pixels.push_back(Pixel2(i%w, i/w, px));
+            ++i;
+        }
+
+        if (allocate_IM)
+            roidata.aux_image_matrix = ImageMatrix(roidata.raw_pixels);
+
+        //std::cout << "printing" << std::endl;
+        //roidata.aux_image_matrix.print();
+        //std::cout << "end printing" << std::endl;
```
Comment (sameeul) [inline tests/test_main_nyxus.h:75]: we can delete these also :-)

File: tests/test_initialization.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    for(int i = 0; i < pending.size(); ++i){
+        LR& r = roiData[i];
+
+        const ImageMatrix& im = r.aux_image_matrix;
+
+        ASSERT_TRUE(im.height > 0);
+        ASSERT_TRUE(im.width > 0);
+    }
+
```
Comment (sameeul) [inline tests/test_initialization.h:36]: We will also probably need `freeTrivialRoisBuffers (pending);` just to be on the safe side of memory leak

File: tests/test_download_data.cc [hunk 3]
```diff
@@ -4,11 +4,15 @@ using namespace std;
 
 int get(const string& url, const string& filename){
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply tests/test_download_data.cc:5]: Not anymore. I removed these files

File: tests/test_main_nyxus.h (hunk 1 above)
Comment (JesseMckinzie) [reply tests/test_main_nyxus.h:75]: Removed

File: tests/test_main_nyxus.h [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+#include <src/nyx/parallel.h>
+#include "test_dsb2018_data.h"
+#include "test_data.h"
+
+namespace Nyxus
+{
+    extern void init_label_record_2(LR& lr, const std::string& segFile, const std::string& intFile, int x, int y, int label, PixIntens intensity, unsigned int tile_index);
+    extern void update_label_record_2(LR& lr, int x, int y, int label, PixIntens intensity, unsigned int tile_index);
+    extern void allocateTrivialRoisBuffers(const std::vector<int>& Pending);
+
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply tests/test_main_nyxus.h:13]: Added

[the remark this answers is not in the document]
Comment (sameeul) [reply tests/test_main_nyxus.h]: Hi Jesse, what I meant was to add those two declarations to `globals.h` and then add `globals.h` as an include in this file.  The functions `init_label_record_2` and `update_label_record_2` are already declared in `globals.h`. Sorry for the lack of clarity in my previous comment. I think this prevents from extra declaration of these functions in the test code and use the declarations from the source.

File: tests/test_initialization.h [hunk 5]
```diff
@@ -11,6 +11,9 @@
 #include "../src/nyx/environment.h"
 #include "../src/nyx/globals.h"
 
+using Nyxus::allocateTrivialRoisBuffers;
+using Nyxus::freeTrivialRoisBuffers;
```
Comment (sameeul) [inline tests/test_initialization.h:15]: I don't think we need these. I removed them and compiled. The only think I had to add was `#include "../src/nyx/globals.h" ` in `test_main_nyxus.h`

**Review by sameeul — APPROVED** [review]

**Pull request merged**